### PR TITLE
SM Iteration 4 Methods Assigned to Me

### PR DESCRIPTION
**Pull request by steve-meyers**
- Add most_accurate_team method and test to StatTracker class
- Add least_accurate_team method and test to StatTracker class
- Add goals_by_season and shots method to game_team_collection
- Add accuracy_ratio method to game_team_collection
- Add accuracy_ratio method call to season_stats hash

### CODE REVIEW

File: lib/game_team_collection.rb [hunk 1]
```diff
@@ -33,11 +33,26 @@ def tackles(season_info, team_id)
     games.sum { | game | game.tackles}
   end
 
+  def shots(season_info, team_id)
+    games = season_info.find_all { | game | game.team_id == team_id}
+    games.sum { | game | game.shots}
```
Comment (Shock253) [inline lib/game_team_collection.rb:58]: there are two enumerables here, can we get these combined into one?

File: lib/game_team_collection.rb [hunk 2]
```diff
@@ -33,11 +33,26 @@ def tackles(season_info, team_id)
...
+  end
+
+  def goals_by_season(season_info, team_id)
+    games = season_info.find_all { | game | game.team_id == team_id}
+    games.sum { | game | game.goals}
+  end
```
Comment (Shock253) [inline lib/game_team_collection.rb:64]: Same problem here, two enumerables is going to be very slow

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -24,6 +28,10 @@ def team_collection
     TeamCollection.new(@team_path)
   end
 
+  def game_team_collection
+     GameTeamCollection.new(@game_team_path)
+  end
```
Comment (Shock253) [inline lib/stat_tracker.rb:33]: every time this method is called it creates a new GameTeamCollection, can we make this into an instance variable

File: test/game_team_collection_test.rb [hunk 4]
```diff
@@ -43,4 +43,9 @@ def test_it_can_calculate_season_stats
     skip
   end
 
+  def test_it_can_get_goals_by_season
+    season_info = @game_team_collection.games_by_season(@game_collection, "20132014")
+    assert_equal 220, @game_team_collection.goals_by_season(season_info, 6)
+  end
```
Comment (Shock253) [inline test/game_team_collection_test.rb:57]: Something to note, why is the season id in a string? if this test is passing it's not accurately representing the actual behavior

**Review by Shock253 — CHANGES_REQUESTED** [review]
The only thing I'd like to change here is the testing in game_team_collection, I'd like to make sure the data types are correct

File: test/game_team_collection_test.rb (hunk 4 above)
Comment (steve-meyers) [reply test/game_team_collection_test.rb:57]: I think the spec harness requires the season as a string.

Comment (mlynch5187) [reply test/game_team_collection_test.rb:57]: Yeah season_id = string

**Pull request closed without merging**